src/qinterface/protected.cpp

```diff
@@ -144,4 +146,41 @@ void exp2x2(complex* matrix2x2, complex* outMatrix2x2) { _expLog2x2(matrix2x2, o
 
 void log2x2(complex* matrix2x2, complex* outMatrix2x2) { _expLog2x2(matrix2x2, outMatrix2x2, false); }
 
+bool isUnitary2x2(const complex* mtrx)
```

Comment:
How expensive is this operation?  Should this be a configuration flag to enable or disable depending on whether or not non-unitary gates are expected?

Reply:
I'll benchmark it, but this is a constant complexity check, compared to gates that grow in complexity like 2 to the number of qubits.